implementations/rust/ockam/ockam_api/src/influxdb_token_lease/token_lease_refresher.rs

```diff
@@ -0,0 +1,125 @@
+use chrono::DateTime;
+use ockam::{compat::time::now, Address, Mailboxes};
+use ockam_core::{api::Error, AllowAll, DenyAll};
+use ockam_node::Context;
+use std::{cmp::max, sync::Arc, time::Duration};
+use tokio::sync::RwLock;
+
+use crate::{
+    cloud::{CredentialsEnabled, ProjectNodeClient},
+    nodes::InMemoryNode,
+    InfluxDbTokenLease,
+};
+
+// The default timeouts are too high.  Set shorter timeouts, so if
+// the project is unresponsible or connection got lost for some reason
+// (like in a project restart), it recover faster.
+const SECURE_CHANNEL_TIMEOUT: Duration = Duration::from_secs(10);
+const REQUEST_TIMEOUT: Duration = Duration::from_secs(15);
+
+#[derive(Clone)]
+pub struct TokenLeaseRefresher {
+    token: Arc<RwLock<Option<String>>>,
+}
+
+impl TokenLeaseRefresher {
+    pub async fn new(
+        ctx: &Context,
+        node_manager: Arc<InMemoryNode>,
+    ) -> Result<TokenLeaseRefresher, Error> {
```

Comment:
The result type should be the one from the [crate](https://github.com/build-trust/ockam/blob/cf7467f5b9d3707effbfd8c9c9cb2d5edbce7424/implementations/rust/ockam/ockam_api/src/cli_state/error.rs#L5), so you don't have to map every error manually.

Reply:
not sure I understand the differences between all the error types and what is compatible with what 😬 , but  some of the errors that could happen inside this function are `CliStateError` indeed,  but some others aren't.     In general,  what error should a failure to start this here should return?  I'm sure it' isn't a `CliStateError` (altough that could be the underling cause),   but really don't know between `ockam_core:Error `or `ockam_api::Error` ,  used the latter as it's on the ockam_api' create and seems the error returned on other modules here.

Reply:
Yeah good point... I see that we are using all three all over the crate, so I guess it makes sense to leave it as it is now.